ladybug/windrose.py

```diff
@@ -0,0 +1,953 @@
+
+        # Calculate regular mesh
+        mesh_array, color_array = WindRose._compute_colored_mesh_array(
+            self.data, self.bin_vectors, ytick_num_mesh, min_bar_radius, max_bar_radius,
+            self.show_stack)
+
+        mesh_array += zero_mesh_array
+        color_array += zero_color_array
+
+        mesh = Mesh2D.from_face_vertices(mesh_array, purge=True)
+        mesh.colors = color_array
+
+        return mesh.scale(self._wind_radius).move(self._center_point2d)
+
+    @property
+    def all_boundary_circles(self):
+        """Get a list of Arc2Ds for the boundary of the plot."""
+        return self._compass.all_boundary_circles
+
+    @property
+    def inner_boundary_circle(self):
+        """Get an Arc2D for the inner boundary of the plot."""
+        return self._compass.inner_boundary_circle
+
+    @property
+    def orientation_lines(self):
+        """Orientation lines for windrose as a LineSegment2D list."""
+        lines = WindRose._xtick_radial_lines(self._bin_vectors)
+        return [line.scale(self._radius).move(self._center_point2d) for line in lines]
+
+    @property
+    def orientation_labels(self):
+        """List of labels corresponding to the orientation angles."""
+        if self._orientation_labels is None:
+            self._orientation_label = self.angles
+        return self._orientation_labels
+
+    @property
+    def orientation_label_points(self):
+        """Get a list of label points from a list of orientation angles between 0 and 360.
+        """
+        return self._compass.label_points_from_angles(
+            self.angles, factor=self.label_point_size_factor)
+
+    @property
+    def orientation_label_ticks(self):
+        """Get a list of Linesegment2Ds from a list of orientation angles between 0 and 360."""
+        return self._compass.ticks_from_angles(self.angles,
+                                               factor=self.label_tick_size_factor)
+
+    @property
+    def frequency_lines(self):
+        """Frequency lines for windrose as Polygon2D lists."""
+        freqs = WindRose._ytick_radial_lines(self._bin_vectors,
+                                             self.legend_parameters.segment_count)
+        return [freq.scale(self._radius).move(self._center_point2d) for freq in freqs]
+
+    @property
+    def _center_point2d(self):
+        """Compute center of windrose plot as Point2D."""
+        return Point2D(self.base_point.x + self._radius,
+                       self.base_point.y + self._radius)
+
+    @property
+    def _radius(self):
+        """Radius of the inner boundary of the windrose plot.
+
+        This number will be a integer multiple of the legend parameter segement count.
+        """
+        ytick_num = self.legend_parameters.segment_count
+        if self.show_zeros:
+            ytick_num += self._ytick_zero_inc
+        return self.frequency_spacing_distance * ytick_num
+
+    @property
+    def _wind_radius(self):
```

Comment:
I also don't understand why you would make this private.

Reply:
Exposed